lib/omniauth/strategies/facebook.rb

```diff
@@ -5,13 +5,42 @@
 require 'uri'
 
 module OmniAuth
+  module Utils
+    class UnknownSignatureAlgorithmError < NotImplementedError; end
+    
+    SUPPORTED_ALGORITHM = 'HMAC-SHA256'
+    
+    def Utils.parse_signed_request(value, secret)
```

Comment:
It is more idiomatic to write these methods as `self. parse_signed_request`. etc.